unstract/connectors/src/unstract/connectors/filesystems/minio/exceptions.py

```diff
@@ -0,0 +1,26 @@
+from unstract.connectors.exceptions import ConnectorError
+
+S3FS_EXC_TO_UNSTRACT_EXC = {
+    "The AWS Access Key Id you provided does not exist in our records": (
+        "Invalid Key (Access Key ID) provided, please provide a valid one."
+    ),
+    "The request signature we calculated does not match the signature you provided": (
+        "Invalid Secret (Secret Access Key) provided, please provide a valid one."
+    ),
+    "[Errno 22] S3 API Requests must be made to API port": (  # Minio only
+        "Request made to invalid port, please check the port of the endpoint URL."
+    ),
+}
+
+
+def handle_s3fs_exception(e: Exception) -> ConnectorError:
+    original_exc = str(e)
+    user_msg = "Error while connecting to S3 / MinIO: "
+    exc_to_append = ""
+    for s3fs_exc, user_friendly_msg in S3FS_EXC_TO_UNSTRACT_EXC.items():
+        if s3fs_exc in original_exc:
+            exc_to_append = user_friendly_msg
+            break
+
+    user_msg += exc_to_append if exc_to_append else f"Error from S3, '{str(e)}'"
```

Comment:
```suggestion
    user_msg += exc_to_append if exc_to_append else str(e)
```

Reply:
`Error from S3` Will get duplicated for unhandled cases.

Reply:
I added it to convey that the error is from the connector's side - you're right duplicating that doesn't look good. Will update the original prefix itself in this scenario